marquez_client/client_wo.py

```diff
@@ -0,0 +1,302 @@
+
+    # Job API
+    def create_job(self, namespace_name, job_name, job_type,
+                   location=None, input_dataset=None,
+                   output_dataset=None, description=None, context=None):
+        log.debug("create_job()")
+
+        MarquezWriteOnlyClient.\
+            _check_name_length(namespace_name, 'namespace_name')
+        MarquezWriteOnlyClient._check_name_length(job_name, 'job_name')
+        MarquezWriteOnlyClient._is_instance_of(job_type, JobType)
+
+        payload = {
+            'inputs': input_dataset or [],
+            'outputs': output_dataset or [],
+            'type': job_type.name
+        }
+
+        if context:
+            payload['context'] = context
+
+        if location:
+            payload['location'] = location
+
+        if description:
+            payload['description'] = description
+
+        return self._backend.put(
+            self._url('/namespaces/{0}/jobs/{1}', namespace_name, job_name),
+            headers=_HEADERS,
+            json=payload
+        )
+
+    def create_job_run(self, namespace_name, job_name, run_id,
+                       nominal_start_time=None,
+                       nominal_end_time=None, run_args=None,
+                       mark_as_running=False):
+        log.debug("create_job_run()")
+
+        MarquezWriteOnlyClient.\
+            _check_name_length(namespace_name, 'namespace_name')
+        MarquezWriteOnlyClient._check_name_length(job_name, 'job_name')
+
+        payload = {}
+
+        payload['id'] = run_id
+
+        if nominal_start_time:
+            payload['nominalStartTime'] = nominal_start_time
+
+        if nominal_end_time:
+            payload['nominalEndTime'] = nominal_end_time
+
+        if run_args:
+            payload['runArgs'] = run_args
+
+        response = self._backend.post(
+            self._url('/namespaces/{0}/jobs/{1}/runs',
+                      namespace_name, job_name),
+            headers=_HEADERS,
+            json=payload)
+
+        if mark_as_running:
+            response = self.mark_job_run_as_started(run_id)
+
+        return response
+
+    def mark_job_run_as_started(self, run_id):
+        log.debug("mark_job_run_as_started()")
+
+        return self.__mark_job_run_as(run_id, 'start')
+
+    def mark_job_run_as_completed(self, run_id):
+        log.debug("mark_job_run_as_completed()")
+
+        return self.__mark_job_run_as(run_id, 'complete')
+
+    def mark_job_run_as_failed(self, run_id):
+        log.debug("mark_job_run_as_failed()")
+
+        return self.__mark_job_run_as(run_id, 'fail')
+
+    def mark_job_run_as_aborted(self, run_id):
+        log.debug("mark_job_run_as_aborted()")
+
+        return self.__mark_job_run_as(run_id, 'abort')
+
+    def __mark_job_run_as(self, run_id, action):
+        log.debug("__mark_job_run_as()")
+
+        MarquezWriteOnlyClient._is_valid_uuid(run_id, 'run_id')
+
+        return self._backend.post(
+            self._url('/jobs/runs/{0}/{1}', run_id, action),
```

Comment:
sorry for forgetting to mention it earlier. We need to add the `?at=` parameter to capture the time this happened instead of when the request was received.
See: https://github.com/MarquezProject/marquez/pull/764

Reply:
Doc updated! https://marquezproject.github.io/marquez/openapi.html#tag/Jobs/paths/~1jobs~1runs~1{id}~1start/post

Reply:
done.